crates/biome_js_semantic/src/semantic_model/builder.rs

```diff
@@ -19,10 +21,10 @@ pub struct SemanticModelBuilder {
     scope_hoisted_to_by_range: FxHashMap<TextSize, usize>,
     bindings: Vec<SemanticModelBindingData>,
     /// maps a binding range to its index inside SemanticModelBuilder::bindings vec
-    bindings_by_range: FxHashMap<TextRange, usize>,
+    bindings_by_range: FxHashMap<(BindingKind, TextRange), usize>,
```

Comment:
We could create a small type `struct` for `(BindingKind, TextRange)`. It's better for creating APIs and documentation.

Reply:
I am planning in an future PR, to use a hack: instead of `(BindingKind, TextRange)` we could just use `TextSize`. For a given node `x`, the value binding could be identifier by `x.range().start()` and the type binding could be identified by `x.range().end()`. This reduces the memory footprint at the minimum.

Reply:
I can integrate the change in this PR if you prefer.

Reply:
I am not really sure I understand the hack :) I'll wait and see